umbraco/Umbraco-CMS
# Add default BlockGrid partial views to disk

This PR adds a migration and an additional step to the BlockGrid example endpoint. 

This will take the default rendering partial views, such as `areas.cshtml`, and add them to the disk, aligning with how the blocklist works. Additionally, when you install the sample we'll also put those partials views on disk, such as `umbBlockGridDemoHeadlineBlock.cshtml`

Unfortunately, there's no good way to do the same with the backoffice custom views. 

I did, unfortunately, have to take a previously internal class (BlockGridSampleHelper), this is to be able to inject it into the controller, the alternative was to inject a service into the `ContentTypeController` just to use it in a constructor, which doesn't seem right. 

## Testing
When testing I did the following 

* Try and run the upgrade - Then ensure that the following partial views are in the tree
	* areas.cshtml
	* default.cshtml
	* items.cshtml
* Create a document type with a blockgrid editor, when prompted if you want to install the sample, do so, then ensure the following views are in the Components folder in the tree (remember to refresh the tree)
	* `umbBlockGridDemoHeadlineBlock.cshtml`
	* `umbBlockGridDemoImageBlock.cshtml`
	* `umbBlockGridDemoRichTextBlock.cshtml`
	* `umbBlockGridDemoTwoColumnLayoutBlock.cshtml`

I also tried packing the solution using `dotnet pack -c Release -o build_output` and created a new site from the resulting template, and everything was there too
